src/connect.ts

```diff
@@ -0,0 +1,42 @@
+import Url from 'url-parse';
+
+import log from './log';
+import { InitBackend, Options } from './types';
+import XAPI from './xapi';
+
+/**
+ * Connect to an XAPI endpoint.
+ *
+ * @example
+ * const xapi = connect('ssh://host.example.com:22');
+ *
+ * @param {string} url - Connection specification.
+ * @param {Object} [options] - Connect options.
+ * @param {string} [options.host] -
+ *     Hostname to connect to.
+ * @param {string} [options.username] -
+ *     Username to authenticate with if protocol requires authentication.
+ * @param {string} [options.password] -
+ *     Password to authenticate with if protocol requires authentication.
+ * @param {string} [options.loglevel] -
+ *     Set the internal log level.
+ * @return {XAPI} - XAPI interface connected to the given URI.
+ */
+export default function connect(url: string, options: Options, initBackend: InitBackend) {
+  if (arguments.length === 1 && typeof url === 'object') {
```

Comment:
this doesnt typescript well at all, the arguments.length == 1 overload doesnt typecheck.

The shortest thing I could find that works is

```
interface Connect {
  (options: object): number;
  (url: string, options: object): number;
}

const connect: Connect = (...args: any[]) => {
  return args.length;
};

connect('str', {});
connect({});
```